training/cifar/cifar10_deepspeed.py

```diff
@@ -243,11 +260,65 @@ def create_moe_param_groups(model):
 # 1) Distributed model
 # 2) Distributed data loader
 # 3) DeepSpeed optimizer
+ds_config = {
+  "train_batch_size": 16,
+  "steps_per_print": 2000,
+  "optimizer": {
+    "type": "Adam",
+    "params": {
+      "lr": 0.001,
+      "betas": [
+        0.8,
+        0.999
+      ],
+      "eps": 1e-8,
+      "weight_decay": 3e-7
+    }
+  },
+  "scheduler": {
+    "type": "WarmupLR",
+    "params": {
+      "warmup_min_lr": 0,
+      "warmup_max_lr": 0.001,
+      "warmup_num_steps": 1000
+    }
+  },
+  "gradient_clipping": 1.0,
+  "prescale_gradients": False,
+  args.dtype: {
+      "enabled": True,
+      "fp16_master_weights_and_grads": False,

```

Comment:
All these loss scaling options are only applicable to fp16, so can we make this conditional?

Reply:
Now BF16 and FP16 config are seperated.  This also allows choosing fp32 as training datatype.